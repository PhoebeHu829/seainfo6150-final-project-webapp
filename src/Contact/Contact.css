.contact_page {
  display: flex;
  justify-content: flex-start;
  margin-left: 100px;
  align-items: center;
  background-image: url('https://img1.mashed.com/img/gallery/the-real-difference-between-regular-popcorn-and-movie-theater-popcorn/intro-1592507709.jpg');
  background-repeat: no-repeat;
  background-position: right bottom;
}

.contact_page .feedback {
  height: 600px;
}

.contact_page .feedback h2 {
  color: rgb(128, 26, 0);
}

.contact_page form {
  margin-top: 50px;
  padding: 20px;
  box-shadow: 10px 10px 50px rgb(92, 95, 92);
  border-radius: 30px;
  font-family: 'Patrick Hand', cursive;
}

.contact_page .form {
  display: grid;
  grid-column-gap: 20px;
  width: 30%;
}

.basicInfo {
  display: grid;
  grid-column-gap: 20px;
}

.basicInfo div {
  margin-top: 10px;
  margin-bottom: 10px;
}

.contact_page input {
  height: 20px;
}

.contact_button {
  margin-top: 20px;
  margin-bottom: 30px;
  width: 200px;
  height: 50px;
  font-family: 'Patrick Hand', cursive;
  font-size: 12px;
  border-radius: 50px;
  border-width: 0px;
  background-color: rgb(58, 5, 156);
  color: azure;
}

.contact_text {
  font-family: 'Orelega One', cursive;
  margin-left: 100px;
  margin-right: 50px;
  width: 500px;
  color: rgb(128, 26, 0);
  background-color: white;
  border-radius: 5px;
  opacity: 0.7;
  padding: 5px;
}

.contact_text p {
  padding: 10px;
}

.error_report svg {
  width: 50px;
  height: 50px;
  margin-right: 10px;
}

.error_report p {
  font-size: 25px;
}

.contact_page .feedback {
  display: grid;
  font-family: 'Patrick Hand', cursive;
}
