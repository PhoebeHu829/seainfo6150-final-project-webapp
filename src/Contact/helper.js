export const emailValidate = (email) => {
  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return re.test(String(email).toLowerCase());
};

export const phoneNumberValidate = (number) => {
  const re_phone = /^[\]?[(]?[0-9]{3}[)]?[-\s\]?[0-9]{3}[-\s\]?[0-9]{4,6}$/im;
  return re_phone.test(String(number));
};
