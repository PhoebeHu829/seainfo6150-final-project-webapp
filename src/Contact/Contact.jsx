import React from 'react';
import './Contact.css';


const Contact = () => {
  return (
    <div className="contact_page">
      <form className='form'>
        <div className="basicInfo">
          <div>
            <div>
              <label className='contact'>Prefix</label>
              <select id="contactDropdown">
                <option>Mr</option>
                <option>Ms</option>
                <option>Mrs</option>
              </select>
            </div>

            <label> First Name:</label>
            <input type="text" name="myText" className="contactText" required />
            <label> Last Name:</label>
            <input type="text" name="myText" className="contactText" required />
          </div>
          <div>
            <label> Email Address:</label>
            <input type="text" name="myText" className="contactText" required />
          </div>
          <div>
            <label> Mobile Phone: </label>
            <input type="text" name="myText" className="contactText" required />
          </div>
        </div>

        <div >
          <label htmlFor="contactDropdownId">Contact Preference</label>
          <select className="contactDropdown">
            <option>Email</option>
            <option>TelePhone</option>
          </select>
        </div>

        <div className='overview'>
          <label htmlFor="contactDropdownId">Available hours:</label>
          <div><input type="radio" name="overview_rate" id="nice" value="4" required /><label htmlFor="nice"> Morning</label></div>
          <div><input type="radio" name="overview_rate" id="good" value="3" /><label htmlFor="good"> Afternoon</label></div>
          <div><input type="radio" name="overview_rate" id="good" value="3" /><label htmlFor="good"> Evenings</label></div>
        </div>

        <label htmlFor="myTextId">Where you hear from us:</label>
        <input type="text" name="myText" className="contactText" required />
        <input type="submit" value="Submit" className='contact_button' />
      </form>

      <div className='contact_text'>
        <h1> Thank You Join Us!</h1>
        <p>
          Join our monthly movie membership and make FREE online reservations in advance.
          Watch movies in any format, including Dolby Cinema, IMAX®, RealD® 3D and more.
          Plus, indulge in the premium perks of AMC Stubs Premiere, including 10% back on food & drink purchases.
          Reserve a ticket as soon as you join.
        </p>
      </div>
    </div >
  )
}

export default Contact;
