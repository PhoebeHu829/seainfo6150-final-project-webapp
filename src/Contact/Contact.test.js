import React from 'react';
import Contact from './Contact.jsx';

it('renders the Contact component correctly', () => {
  const { container } = render(<Contact user={''} setUser={() => {}} />);
  expect(container).toMatchSnapshot();
});
