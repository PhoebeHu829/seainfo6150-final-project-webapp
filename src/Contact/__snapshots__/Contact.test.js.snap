// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Contact component correctly 1`] = `
<div>
  <div
    class="contact_page"
  >
    <form
      class="form"
    >
      <div
        class="basicInfo"
      >
        <div>
          <div>
            <label
              class="contact"
            >
              Prefix
            </label>
            <select
              id="contactDropdown"
            >
              <option
                value="Mr"
              >
                Mr
              </option>
              <option
                value="Ms"
              >
                Ms
              </option>
              <option
                value="Mrs"
              >
                Mrs
              </option>
            </select>
          </div>
          <div>
            <label>
               First Name:
            </label>
            <input
              class="contactText"
              name="myText"
              required=""
              type="text"
            />
          </div>
          <label>
             Last Name:
          </label>
          <input
            class="contactText"
            name="myText"
            required=""
            type="text"
          />
        </div>
        <div>
          <label>
             Email Address:
          </label>
          <input
            class="contactText"
            name="myText"
            required=""
            type="text"
          />
        </div>
        <div>
          <label>
             Mobile Phone: 
          </label>
          <input
            class="contactText"
            name="myText"
            required=""
            type="text"
          />
        </div>
      </div>
      <div>
        <label
          for="contactDropdownId"
        >
          Contact Preference
        </label>
        <select
          class="contactDropdown"
        >
          <option
            value="Email"
          >
            Email
          </option>
          <option
            value="TelePhone"
          >
            TelePhone
          </option>
        </select>
      </div>
      <div
        class="overview"
      >
        <label
          for="contactDropdownId"
        >
          Available hours:
        </label>
        <div>
          <input
            id="nice"
            name="overview_rate"
            required=""
            type="radio"
            value="morning"
          />
          <label
            for="nice"
          >
             Morning
          </label>
        </div>
        <div>
          <input
            id="good"
            name="overview_rate"
            type="radio"
            value="afternoon"
          />
          <label
            for="good"
          >
             Afternoon
          </label>
        </div>
        <div>
          <input
            id="good"
            name="overview_rate"
            type="radio"
            value="evenings"
          />
          <label
            for="good"
          >
             Evenings
          </label>
        </div>
      </div>
      <label
        for="myTextId"
      >
        Where you hear from us:
      </label>
      <input
        class="contactText"
        name="myText"
        type="text"
      />
      <input
        class="contact_button"
        type="submit"
        value="Submit"
      />
    </form>
    <div
      class="contact_text"
    >
      <h1>
         Thank You Join Us!
      </h1>
      <p>
        Join our monthly movie membership and make FREE online reservations in advance. Watch movies in any format, including AAA Cinema, IMAX, Real 3D and more. Plus, indulge in the premium perks of AAA Stubs Premiere, including 10% back on food & drink purchases. Reserve a ticket as soon as you join.
      </p>
    </div>
  </div>
</div>
`;
